src/auro-toast.js

```diff
@@ -81,30 +103,82 @@ export class AuroToast extends LitElement {
     const toastContainer = this.shadowRoot.querySelector('#toastContainer');
     toastContainer.className = 'hidden';
 
-    // Wait until fade out completes to set visible to false
     setTimeout(() => {
-      this.visible = false;
+      this.closeToast();
     }, FADE_OUT_DURATION);
   }
 
-  updated() {
-    if (this.visible) {
+  /**
+   * @private
+   * @returns {void}
+   */
+  closeToast() {
+    clearTimeout(this.fadeOutTimer);
+    this.visible = false;
+    this.dispatchEvent(new CustomEvent('onToastClose', {
+      bubbles: true,
+      composed: true,
+      detail: this
+    }));
+  }
+
+  /**
+   * For mobile, set the onclick function so the toast can be dismissed if it is tapped on anywhere inside the toast.
+   * @private
+   * @returns {void}
+   */
+  setOnClick() {
+    const mobileBreakPoint = 767;
+    if (window.innerWidth < mobileBreakPoint) {
+      this.onclick = () => {
+        this.fadeOutToast();
+      };
+    }
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.setOnClick();
+  }
+
+  updated(changedProperties) {
+    if (changedProperties.has('variant')) {
+      clearTimeout(this.fadeOutTimer);
+    }
+    // do not auto dismiss for error toasts
+    if (this.visible && this.variant !== 'error') {
       this.fadeOutTimer = setTimeout(() => {
         this.fadeOutToast();
       }, TIME_TIL_FADE_OUT);
     }
   }
 
   render() {
-    return this.visible
-      ? html`
-      <div id="toastContainer">
-        <slot></slot>
-        <button class="toastButton" @click="${this.handleOnClose}">
-          ${this.svg}
-        </button>
-      </div>
-      `
+    let iconHtml = html``;
+
+    switch (this.variant) {
+      case undefined:
+      case null:
+        // default toast uses information icon
+        iconHtml = this.generateIconHtml(information.svg);
+        break;
+      case "error":
+        iconHtml = this.generateIconHtml(error.svg);
+        break;
+      case "success":
+        iconHtml = this.generateIconHtml(success.svg);
+        break;
+      default:
+        break;
+    }
+
+    return this.visible ? html`<div id="toastContainer">
+      ${iconHtml}
+        <div class="message"><slot></slot></div>
+          <button class="closeButton" aria-label="closeToast" @click="${this.handleOnClose}">
+            ${this.svg}
+          </button>
+      </div>`
```

Comment:
Intentations are indorrect. Line 177 opens and closes a DIV. `<button>` is a sibling, not a child. 

```suggestion
    return this.visible ? html`<div id="toastContainer">
      ${iconHtml}
        <div class="message"><slot></slot></div>
        <button class="closeButton" aria-label="closeToast" @click="${this.handleOnClose}">
          ${this.svg}
        </button>
    </div>`
```

Reply:
@blackFalcon, what's your take on adding an auto linting commit hook?

Reply:
That's a debate at the Generator level. If you have ideas, submit them there.